kernel/apic/src/lib.rs

```diff
@@ -38,6 +36,20 @@ const _APIC_TIMER_MODE_TSC_DEADLINE:   u32 = 0b10 << 17;
 pub const LOCAL_APIC_LVT_IRQ:          u8  = 0x22;
 
 
+/// A unique identifier for a Local APIC, i.e., a CPU on x86_64.
+#[derive(
+    Clone, Copy, Debug, Display, PartialEq, Eq, PartialOrd, Ord,
+    Hash, Binary, Octal, LowerHex, UpperHex,
+)]
```

Comment:
the type isn't actually opaque since you can access the inner value. It's opaque in the sense that you cannot instantiate one yourself because the inner value is hidden. But this raises a good point; I will clarify the docs.

There's no harm in implementing `Ord`, in fact it's necessary to allow it to be used in an ordered/sorted type like `BTreeMap`.

Reply:
Actually the docs don't use the word opaque, so they're already fine.